library/spit_results.py

```diff
@@ -288,6 +288,45 @@ def classify(key, fact_names, classifications):
     return result
 
 
+def raw_output_present(fact_names, host_vars, this_fact, this_var, command):
+    """Basic sanity checks for processing an Ansible raw command.
+
+    :param fact_names: the facts to be collected
+    :param host_vars: all variables collected for a host
+    :param this_fact: the name of the fact we are processing
+    :param this_var: the name that Ansible has for our output
+    :param command: the command that was run
+
+    :returns: a tuple of
+        (None or error dict, None or raw command output).
+        The error dict is suitable for inclusion in the rho output
+        dictionary. There will not be both errors and raw command
+        output. If raw command output is returned, it will have
+        fields 'rc' and 'stdout_lines' or 'results'.
+
+    Usage:
+        err, output = raw_output_present(...)
+        if err is not None:
+            return err
+
+        ... process output ...
+    """
+
+    if this_fact not in fact_names:
+        return {}, None
+
+    if this_var not in host_vars:
+        return {this_fact: 'Error: "{0}" not run'.format(command)}, None
+
+    raw_output = host_vars[this_var]
+
+    if (('rc' not in raw_output or
+         'stdout_lines' not in raw_output) and 'results' not in raw_output):
+        return {this_fact: 'Error: "{0}" not run'.format(command)}, None
```

Comment:
Do we want to differentiate between this and line 319 in the output. Maybe something like `Error: data from "{0}" command could not be obtained`? Might help with debugging, thoughts?

Reply:
I like that! Done.